### ISSUE TITLE
Show a country level ratio

### ISSUE THREAD

**Issue by icco** [rtcovidlive/covid-model#6]
It'd be really nice to also see an Rt for the entire US.

**Comment by k-sys** [comment]
Thank you for the feedback.

(1) We haven't done this yet because we've been clear that you can't manage the country as a whole as much as you can manage local infections, but this argument is true at a state level – which is why it's better to see county-level metrics. I'm a bit reluctant to add it because it's not a metric we can actually manage.

(2) It's actually a difficult problem to solve, because testing regimes/standards across states are actually quite different. We've brainstormed a bit about different approaches and we'll see where we get.

On balance, I'm leaning towards adding something like this if we find it because I think intellectually it's an interesting metric even if it's not clearly actionable.

**Comment by icco** [comment]
Yeah, I'm just kind of curious. I'm not someone making policy or anything, I just wanna know how the country is doing as a whole.

**Comment by benmarten** [comment]
Second that!
While the management of the pandemic might require different approaches on all levels of the government, we should at least be able to be consistent when comparing countries.
The US is still a single country, so comparing it to Other countries would still be helpful. I.e. compare apples to apples.
For example the EU is a federation of states, while Germany is a federal republic, similar to the US. The sheer size might not be a factor, since the us is also often compared to China, Russia or Brazil!

**Comment by michaelosthege** [comment]
Implementation note:

There are countries where the sum of all states is less than the numbers for the entire country (in Germany b/c of data protection, for example). Therefore, it may be easiest to include a `"all"` region in the dataframes and handle it like any other region.

**Comment by tywmick** [comment]
Would a weighted average of all the states' <var>R<sub>t</sub></var> estimates, weighted by each state's population, be a very rough but acceptable proxy for the country's <var>R<sub>t</sub></var> overall? Or would that be wildly inaccurate?

Not to add that to R<sub>t</sub> Live's homepage—the reason I ask is because I'm building a little [project](https://github.com/tywmick/rt-simulator) visualizing the consequences of various <var>R<sub>t</sub></var> values over the next four months. I'm fetching your [calculated <var>R<sub>t</sub></var> values](https://d14wlfuexuxgcm.cloudfront.net/covid/rt.csv) to let users feed their state's current <var>R<sub>t</sub></var> estimate (including confidence intervals) and current average daily new cases into the simulation, but I feel like it'd be interesting to let them fill in a current estimate for the United States as a whole, as well.

**Comment by michaelosthege** [comment]
I don't think a weighted average can get you there. If at all, it would have to be by number of tests or positives. 

The main limitation about summarizing for the entire country is to do the sum over the performed and positive tests. There is often at least one region that has a gap in the data on a particular day. And nansum will give you a number that's too low.. First you have to make sure that 0 actually means 0 and `nan` is used for "we don't know" or "obviously wrong". 

For Germany (https://rtlive.de) we first apply forecasting (see PRs here) to fill gaps before doing the sum.
If I had the time, I'd already done it for the US) - just adding another region "all". But with >50 regions there's a lot more diversity of data artifacts (OK, for example) that mandates a closer look.
